src/app/workflow/view/view.component.ts

```diff
@@ -89,6 +93,25 @@ export class ViewWorkflowComponent extends View implements OnInit {
     });
   }
 
+  snapshotVersion(): void {
+    // Create a new version temporarily with the frozen bit set. We assume the version is
+    // not already a snapshot since the UI controls shouldn't be available for
+    // snapshotted versions.
+    var snapshot = { ...this.version };
+    snapshot['frozen'] = true;
+    this.workflowsService.updateWorkflowVersion(this.workflow.id, [snapshot]).subscribe(
+      result => {
```

Comment:
`result` should be an array of updated workflow versions.

1. Name should reflect it
2. Is it possible to use those versions to update the store rather than calling refreshWorkflow below?

Reply:
1. Taking a quick look at the ws it seems like you get all workflowVersions back either way so I changed to `workflowVersions`.
2. I certainly imagine so, is that the usual pattern? I copied the modify/refresh pattern from the versionModal.

Reply:
Not sure if it's the usual pattern, but it seems like it should be. My only concern is that there is something in the workflow itself that changes when you update a version (just speculating), but hopefully not.